app/Services/InviteService.php

```diff
@@ -2,32 +2,73 @@
 
 namespace App\Services;
 
+use Firebase\JWT\JWT;
 use App\Models\Invite;
 use App\Mail\InviteMail;
 use Illuminate\Support\Facades\Mail;
+use App\Constants\Invite as InviteConstant;
 
 class InviteService
 {
-    public function generateToken(): string
+    public function generateUrl($token, array $user): String
     {
-        return bcrypt(now()->addDays(5)->toDateTimeLocalString());
+        return config('frontend.url') . '/register/' . $token . '?email=' . $user['email'] . '&name=' . urlencode($user['name']);
     }
 
-    public function processInvite(array $validatedInviteUser): mixed
+    public function generateToken(array $user): string
     {
-        $token = $this->generateToken();
+        $issuedAt = time();
 
-        $url = config('frontend.url') . '/register/' . $token . '?email=' . $validatedInviteUser['email'] . '&name=' . urlencode($validatedInviteUser['name']);
+        $expirationTime = $issuedAt . InviteConstant::DEFAULT_EXIPIRE_AT;
+
+        $payload = array(
+            'name' => $user['name'],
+            'email' => $user['email'],
+            'iat' => $issuedAt,
+            'exp' => $expirationTime,
+        );
+
+        $token = JWT::encode(
+            $payload,
+            config('jwt.secret_key'),
+            config('jwt.algorithm')
+        );
+
+        return $token;
+    }
+
+    public function processInvite(array $validatedInviteUser)
+    {
+        $token = $this->generateToken([
+            'name' => $validatedInviteUser['name'],
+            'email' => $validatedInviteUser['email']
+        ]);
+
+        $url = $this->generateUrl(
+            $token,
+            [
```

Comment:
@sabinshrestha8 Directly pass the $validatedInviteUser

Reply:
https://github.com/proshore/ProshoreAssetManagement-Backend/pull/12/commits/b5c591c41b54e3c9a63f101fd61e5f77381d5909